### ISSUE TITLE
xUnit support

### ISSUE THREAD

**Issue by wahmedswl** [axodox/AxoTools#4]
Hi, 
   I am using xUnit for Coverage, is it supported cuz AxoCover not able to detect any Tests within solution. Is there any configuration which need to be tweaked?

Thanks

**Comment by axodox** [comment]
Hi,

Unfortunately xUnit is not yet supported, however I plan to add support to both NUnit and xUnit as soon as I finish the main features I planned. So I would say it is likely to be added in a month or so, depending on how much free time I have - and how many contributors join ;).

Cheers,
Peter

**Comment by Caldas** [comment]
I hope you can get support to Xunit also 😃

**Comment by axodox** [comment]
I looked into it, and decided to add support through the visual studio testadapter implementations of xUnit and NUnit. This way the test framework specific work will be minimal, and support should start with a well rounded set of features. In addition, if a framework has support for visual studio test platform, it will be trivial to add. Please note however, that OpenCover imposed platform limitations will still apply (so no UWP, Xamarin etc.).

Based on my current results I think it will take around a month to have a releasable implementation (not so much work, but so little free time). However if you are willing to beta test it, I will upload a version when it reaches a usable state for you to check.

**Comment by Caldas** [comment]
I can start to test as soon you release it as beta

Em 12 de fev de 2017 3:11 PM, "axodox" <notifications@github.com> escreveu:

**Comment by axodox** [comment]
Should you be very eager to try it, here is an early version for you. 

Features:
- Support for many kinds of xUnit and NUnit tests, including data driven theories
- This support is for the "classic" desktop .Net applications and services
- .Net core, UWP, Xamarin etc. is NOT supported and as the support of these will require a lot of changes in OpenCover, it is not likely in near future, however you can also put code in portable class libraries, which can help somewhat
- Can handle vstest / xUnit / NUnit together and execute / cover them by one click (no need to select)
- There is internal support to VS test adapters, so I can add easily more test frameworks later (e.g. SpecFlow), but as in this case, it might require a little tweaking in the adapter, still saves tons of time
- No console output stream reading anymore, but a good TCP based connection directly to the test runner, which results in faster, more reliable and rich data during test running, now you see what test is running right now, you don't have to wait until the end to see stack traces etc.

Known issues:
- The test settings are not supported in this version (the menu is present, but it does nothing)
- Debugging of NUnit, xUnit tests will only work, if you install the adapters for visual studio, later I will add native support for debugging tests, which will not rely on visual studio understanding them
- All your tests will run on one STA thread in an x86 process, these all will be configurable soon (directly and through runsettings) - so you can runt tests in parallel, x64, MTA as you need
- Not all test types are supported yet, don't worry I will add them later
- This is especially true to cover by test feature, for which I will need to add some changes to OpenCover too
- Aborting the tests might not work properly, leave processes running and could even hang Visual Studio, so I would not use it much
- Old test runners are removed, this might remain like this, or I might add them back temporarily to ease transition, in case regressions present in new code (I mean on release, not there are a number of regressions, no question)
- In general I have replaced the old test runner - which worked by launching vstest console and reading back the reports as well as the console output to check progress to a solution - to a new one where I use my own test runner and connect to it through a WCF service.
- So a lot of changes, not tested, work in progress, things might very well get funky.

[AxoCover-20170214.2.zip](https://github.com/axodox/AxoTools/files/778679/AxoCover-20170214.2.zip)

**Comment by Caldas** [comment]
I installed your version:

![image](https://cloud.githubusercontent.com/assets/938045/23005395/7322dd4a-f3e3-11e6-8841-91cf5537d049.png)

But I don't know why it's telling me that there are no tests:

![image](https://cloud.githubusercontent.com/assets/938045/23005412/7ef58c94-f3e3-11e6-96a2-225612598a7e.png)

**Comment by axodox** [comment]
Thanks for trying it out. As I mentioned above in my release notes _.Net Core is not yet supported_, because of OpenCover limitations. But now I have found [this](https://github.com/OpenCover/opencover/issues/601). It is suggested that there is a workaround around this limitation, so there is hope. However based on the description you will need to change some compilation settings for it work too (to get old style PDB - this is worked on in OpenCover) - and of course other changes will be needed for AxoCover too.

I will first need to add the .Net Core discoverer of xUnit. And then I will need to make the AxoCover.Runner .Net Core compatible, I have not worked with .Net Core yet, but I assume there will be some work involved, to get it work with my runner and OpenCover together. 

I will try out what happens if load the .net core version of the test discovery / runner module of xUnit. And if I can get it to run, I will add it soon. If needed I will port my runner to .Net Core, based on the API's used there should be no problem. Otherwise - if it is more complex, or I feel the OpenCover current support poor - I think I will try to ship the XUnit + NUnit support for the full .Net Framework first and then add the support for .Net Core.

**Comment by axodox** [comment]
I just see it is the same discoverer / executor, so it might be I skip the .Net Core projects because of their project type - I only search tests in .Net projects which have references to specific test libraries. I will look into it in the evening if I have the time.

**Comment by wahmedswl** [comment]
Hi,
   I have tried in VS2015, .NET Project and its working fine.

![image](https://cloud.githubusercontent.com/assets/15020561/23021318/349c380c-f46d-11e6-988b-613f1171a933.png)

Thanks for your quick work done on this issue.

**Comment by wahmedswl** [comment]
Hi, I have enabled ```Cover after each build``` and i got following output which show exceptions also
```
Coverage executor has started.
INFO Executing tests...
INFO Running executor: Xunit.Runner.VisualStudio.TestAdapter.VsTestRunner...
INFO [xUnit.net 00:00:01.2410339]   Starting:    TestAndPerf
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_CombinePath (0af22a1d4e9972e90ec2f5656d11282a2801608b)
INFO Passed   TestAndPerf.Test.TestSequelSql.Test_SequelSql_IdxAndStrip (1953e4fa7a04ec70e2b9f3d56d3d2840b3d3a510)
INFO Passed   TestAndPerf.Test.TestExtensions.Test_Extensions_XElement_Val (1152703e3bfd664dc84d15ddb7099d69a74f6ab5)
INFO Passed   TestAndPerf.Test.TestDatabase.Test_Sql_Paged (127f7f24db01aa24fb53d1ba1b7ffb280b30b451)
INFO Passed   TestAndPerf.Test.TestExtensions.Test_Extensions_AddUnique (220fd4e06ad766e68802335c9dea9ec5d3278c27)
INFO Passed   TestAndPerf.Test.TestExtensions.Test_Extensions_String_Int (24d0edf2583b9da1a32746ed729f266f5e690d59)
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_Enc (10f9e3a988bb0fceda394077b9cd005c77319ee2)
INFO Passed   TestAndPerf.Test.TestExtensions.Test_Extensions_String_Float (36428dd999fa33d24b5f03e53f9ce766439b7945)
INFO Passed   TestAndPerf.Test.TestExtensions.Test_Extensions_String_Double (3bc846f6965bd4c9fc1089183fb1c7ff3c2331b8)
INFO Passed   TestAndPerf.Test.TestExtensions.Test_Extensions_String_Replace (4cf869bf285a1bfa7d638ca9cf6608f327ea9656)
INFO Passed   TestAndPerf.Test.TestExtensions.Test_Extensions_Array_Val (4fe34742b21a0227dab6caec85bf368f4dac7f3a)
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_Join (12a447de805d537e0fb2b4f22780fe1ac40b49b6)
INFO Passed   TestAndPerf.Test.TestExtensions.Test_Extensions_String_Bool (54690d31b78a6b234f851f0a5ddb9e07f3632eed)
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_Client (192a3459b5163fb769f431cd3850d1c735e6825f)
INFO Passed   TestAndPerf.Test.TestExtensions.Test_Extensions_Columns (56ec1b198d9bd6626feda36fbcb68a6e0beb28df)
INFO Passed   TestAndPerf.Test.TestExtensions.Test_Extensions_Class_Nil (57e00e592b3e04b1cfda6052837896c1c7931441)
INFO Passed   TestAndPerf.Test.TestExtensions.Test_Extensions_Else (5a267dd68b998f15edf182a6544f475651e17ce5)
INFO Passed   TestAndPerf.Test.TestExtensions.Test_Extensions_String_Slice (5d843a486685c714f62616b0b91d0eaed2eb3e7f)
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_Count (1b4c723550bef746793668f7bcabc80173d91bc5)
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_Redact (284f00935474a34cfb9a9339aa2265666cc4791d)
INFO Passed   TestAndPerf.Test.TestExtensions.Test_Extensions_String_ForwardBracket (6bf42bfcf237df76d76c2c5357047ab4122c6301)
INFO Passed   TestAndPerf.Test.TestExtensions.Test_Extensions_Dictionary_Val (711875de7f807b509b2a6dedc21022dbaa86286e)
INFO Passed   TestAndPerf.Test.TestExtensions.Test_Extensions_String_Compression (934dd340953dcae76ce2236a519a1c8464164f1a)
INFO Passed   TestAndPerf.Test.TestExtensions.Test_Extensions_Column (a59be8035d94dfb7de499d6d5a22f43be83d621b)
INFO Passed   TestAndPerf.Test.TestExtensions.Test_Extensions_String_Long (cc4d4e37bec6d7d041ba3607492edf9334b9f5e0)
INFO Passed   TestAndPerf.Test.TestExtensions.Test_Extensions_EnqueueUnique (dd463341f9e64f3d1244624bd972ac48b4387ea8)
INFO Passed   TestAndPerf.Test.TestExtensions.Test_Extensions_String_BaseUrl (ed12780ba0cfc6d89dfedf2887f325375b0b22a5)
INFO Passed   TestAndPerf.Test.TestExtensions.Test_Extensions_Generic_Map (f2c58d6a048a179f89d77c98f78256ba2e064819)
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_Cache (2eafb81a87e37c9e99acf986e90a0c40d45e569d)
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_Empty (3d192983d0cbf824ab2e5b1aef1152b1f8962355)
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_DB (4d37e14098b021edda1c5527b068289a0b9c79cc)
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_Eval (53a37969fbce0a164788233ae7ab8d529dea7aa7)
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_Decode (5970152dc1d068b18d588d1cd5e674738240f0db)
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_Array (60c097a84f411f06ae5e40c09012ebdf7589eb4f)
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_OkByCfg (78f0caa282fbc998d23284d8030642a41f27994c)
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_Str (7cdbe20a994884724c63b116a25c3f6b366d50b2)
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_SvcName (a713fd5206d0bbc77b1969c18e9b715ce55cab5b)
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_Dump (bbe059ecb770b18dfda7a0d1599bf9b61b5ea88d)
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_CStr (caa71a2a895aaa451af304a2fb21e861dd009dfc)
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_Fmt (d21a5732b6f0b8cc7c651a4b81840c9b812f0b0d)
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_Val (d6ed4a6992188cb1a38e0cc96f3da8348828c431)
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_CSV (e1b54118af4635e7f40503092025b4b6b6497761)
INFO Passed   TestAndPerf.Test.TestLogging.Test_Logger_Cfg (fad7b36cec5f8d32031f433bc193658d533f0b9d)
INFO Passed   TestAndPerf.Test.TestDatabase.Test_Database_Delete (2198afacea30633a279a15b9db8345be994681d0)
INFO Passed   TestAndPerf.Test.TestDatabase.Test_Database_ConnectionString (2f08db926adffb520ac08d586304635585965696)
INFO Passed   TestAndPerf.Test.TestDatabase.Test_Database_ParamIdx (3f50575edb20918688e0071fe24634dac8aa14a8)
INFO Passed   TestAndPerf.Test.TestDatabase.Test_Database_ParamName (4ccc228872428cf11eef34a9622566b4d4219a5e)
INFO Passed   TestAndPerf.Test.TestDatabase.Test_Database_Connection (61e43cf3023d3e5b3d3581da51d5d10d52f86e07)
INFO Passed   TestAndPerf.Test.TestDatabase.Test_Database_Tx (7338eb932384ca335fc27c05aef80da1445adfa0)
INFO Passed   TestAndPerf.Test.TestSequelSql.Test_SequelSql_Select (4aafe8dc23d57793127dd17f0d0ed7b517623828)
INFO Passed   TestAndPerf.Test.TestSequelSql.Test_SequelSql_QueryAndIdx (b4ff4cdb503d0e94ead6fcba975fa5de05ac70c9)
INFO Passed   TestAndPerf.Test.TestSequelSql.Test_SequelSql_ChangeParamsIdx (c1a9495d17e017d2e9cc5fbc62f60b347e5e0985)
INFO Passed   TestAndPerf.Test.TestDatabase.Test_Database_Select (89e4a0bf3e7bbd898e8faaa09ba9ea4a449fb7a2)
INFO Passed   TestAndPerf.Test.TestDatabase.Test_Database_Insert (8f52ec0d730dd400af1e282b47edde1e2ac49e89)
INFO Passed   TestAndPerf.Test.TestDatabase.Test_Database_Page (c83aad1a1420214efe810bad12d8e7da6e451567)
INFO Passed   TestAndPerf.Test.TestDatabase.Test_Database_Update (cb9456e053d98e3df7143505a6016160a2a0f42a)
INFO Passed   TestAndPerf.Test.TestDatabase.Test_Database_TransformCommand (ccaa654dbb50e10bae03a53669d35a2c7f383ede)
INFO Passed   TestAndPerf.Test.TestDatabase.Test_Database_TableName (ddf1c85dea7ca36a5d6f7071aeebd7ddb2df4a8a)
INFO [xUnit.net 00:00:07.0498064]   Finished:    TestAndPerf
INFO Executor finished.
INFO Test execution finished.
Coverage executor has finished.
AxoCover encountered an error:
KeyNotFoundException: The given key was not present in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at AxoCover.Models.CoverageProvider.<>c__DisplayClass10_3.<GetFileCoverage>b__31(IGrouping`2 p)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at AxoCover.Models.CoverageProvider.GetFileCoverage(String filePath)
   at AxoCover.Models.CoverageProvider.<>c__DisplayClass9_0.<GetFileCoverageAsync>b__0()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at AxoCover.Models.CoverageProvider.<GetFileCoverageAsync>d__9.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at AxoCover.LineCoverageAdornment.<UpdateCoverage>d__65.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_0(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

**Comment by wahmedswl** [comment]
Also, when i first run the AxoCover, it generated markers in the left side which shows if i cover some code or not as visible in https://cloud.githubusercontent.com/assets/15020561/23021318/349c380c-f46d-11e6-988b-613f1171a933.png but now i cannot see that. 

And this seems to happens only when i check Exclude test assemblies in Settings.

**Comment by axodox** [comment]
Great find! Now I understand an issue which showed up in the telemetry yesterday. The master branch has these fixed to degree (with a workaround for now, until I have time to create a better fix), but I am at work now, so I will bring the fixes over in the evening.

The two issues: 
- [if you use non-English version of .Net, stack trace cannot be parsed properly](https://github.com/axodox/AxoTools/issues/4)
- [excluding test assemblies breaks coverage visualization in some coses](https://github.com/axodox/AxoTools/issues/8)

**Comment by axodox** [comment]
I have added a fix for what @wahmedswl have found. Also brought over other fixes from the master branch.

[AxoCover-20170216.1.zip](https://github.com/axodox/AxoTools/files/781823/AxoCover-20170216.1.zip)

For .Net Core I started investigation, it seems that the standard version of xunit test adapter will not find the .Net Core tests, and I had trouble loading the .Net Core one into my runner process. I will try more at the weekend, but if I cannot get it working to a degree in an couple of hours I will put it on hold for now, and focus on delivering the full framework version first. If you have time, you are free to grab the code and look into it, if you have the VS SDK installed it builds out-of-the-box.

**Comment by wahmedswl** [comment]
@axodox thanks for your quick response, i really appreciate it.

**Comment by axodox** [comment]
[AxoCover-20170220.1.zip](https://github.com/axodox/AxoTools/files/786848/AxoCover-20170220.1.zip)
 Weekend update with improvements to test settings, aborting, and debugging.

**Comment by axodox** [comment]
Error to be fixed:
```
Application: devenv.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.ServiceModel.CommunicationObjectFaultedException

Server stack trace: 
   at System.ServiceModel.Channels.CommunicationObject.ThrowIfDisposedOrNotOpen()
   at System.ServiceModel.Channels.ServiceChannel.EnsureOpened(TimeSpan timeout)
   at System.ServiceModel.Channels.ServiceChannel.Call(String action, Boolean oneway, ProxyOperationRuntime operation, Object[] ins, Object[] outs, TimeSpan timeout)
   at System.ServiceModel.Channels.ServiceChannelProxy.InvokeService(IMethodCallMessage methodCall, ProxyOperationRuntime operation)
   at System.ServiceModel.Channels.ServiceChannelProxy.Invoke(IMessage message)
   at System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(System.Runtime.Remoting.Messaging.IMessage, System.Runtime.Remoting.Messaging.IMessage)
   at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(System.Runtime.Remoting.Proxies.MessageData ByRef, Int32)
   at AxoCover.Common.Runner.ITestDiscoveryService.TryLoadAdaptersFromAssembly(System.String)
   at AxoCover.Models.DiscoveryProcess.OnServiceStarted(System.Uri)
   at AxoCover.Common.ProcessHost.ServiceProcess.OnOutputDataReceived(System.Object, System.Diagnostics.DataReceivedEventArgs)
   at System.Diagnostics.Process.OutputReadNotifyUser(System.String)
   at System.Diagnostics.AsyncStreamReader.FlushMessageQueue()
   at System.Diagnostics.AsyncStreamReader.GetLinesFromStringBuilder()
   at System.Diagnostics.AsyncStreamReader.ReadBuffer(System.IAsyncResult)
   at System.IO.Stream+ReadWriteTask.InvokeAsyncCallback(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.IO.Stream+ReadWriteTask.System.Threading.Tasks.ITaskCompletionAction.Invoke(System.Threading.Tasks.Task)
   at System.Threading.Tasks.Task.FinishContinuations()
   at System.Threading.Tasks.Task.FinishStageThree()
   at System.Threading.Tasks.Task.FinishStageTwo()
   at System.Threading.Tasks.Task.Finish(Boolean)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(System.Threading.Tasks.Task ByRef)
   at System.Threading.Tasks.Task.ExecuteEntry(Boolean)
   at System.Threading.Tasks.Task.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
```

**Comment by axodox** [comment]
Note to self: when running without coverage, exception glyphs should still be displayed.

**Comment by axodox** [comment]
Added a large amount of changes and fixes:
- Support parametrized, generic and nested test fixtures with NUnit (XUnit support is also in progress)
- Switched to forked and AxoCover improved builds of OpenCover, and NUnit / XUnit adapters for now
- Better support for debugging non-MS tests
- Improved networking behavior with AxoCover.Runner
- Updated support for output directory management

This version is getting close to be feature complete for 1.1 release. I plan to finish up and release changes next week.

[AxoCover-20170312.3.zip](https://github.com/axodox/AxoTools/files/836972/AxoCover-20170312.3.zip)

**Comment by axodox** [comment]
**BUG**: _debug button on test details view can stuck on disabled_
**BUG**: _sometimes after the debugger attached the test runners fails to exit_

**Comment by axodox** [comment]
From now on new versions, including pre-release versions such as these are auto-generated by AppVeyor, and show up as releases on GitHub.

**Comment by wahmedswl** [comment]
Hi, seems like xUnit is having problem. I am using Paket for dependencies and when running AxoCover, i am getting following output

```
Test discovery started.
INFO Loading assembly from C:\Users\wahmed\AppData\Local\Microsoft\VisualStudio\14.0\Extensions\e3f0h5oa.5qr\MSTestAdapter\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.dll...
INFO Assembly loaded.
INFO Loading assembly from C:\Users\wahmed\AppData\Local\Microsoft\VisualStudio\14.0\Extensions\e3f0h5oa.5qr\NUnitAdapter\NUnit3.TestAdapter.dll...
INFO Assembly loaded.
INFO Loading assembly from C:\Users\wahmed\AppData\Local\Microsoft\VisualStudio\14.0\Extensions\e3f0h5oa.5qr\xUnitAdapter\xunit.runner.visualstudio.testadapter.dll...
INFO Assembly loaded.
INFO Discovering tests...
INFO Running discoverer: Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.MSTestDiscoverer...
INFO Checking E:\Development\Production-WS\Infrastructure\TestAndPerf\bin\Debug\TestAndPerf.dll...
INFO Discoverer finished.
INFO Running discoverer: NUnit.VisualStudio.TestAdapter.NUnit3TestDiscoverer...
INFO Checking E:\Development\Production-WS\Infrastructure\TestAndPerf\bin\Debug\TestAndPerf.dll...
INFO NUnit Adapter 3.7.0.0: Test discovery starting
INFO Assembly contains no NUnit 3.0 tests: E:\Development\Production-WS\Infrastructure\TestAndPerf\bin\Debug\TestAndPerf.dll
INFO NUnit Adapter 3.7.0.0: Test discovery complete
INFO Discoverer finished.
INFO Running discoverer: Xunit.Runner.VisualStudio.TestAdapter.VsTestRunner...
INFO Checking E:\Development\Production-WS\Infrastructure\TestAndPerf\bin\Debug\TestAndPerf.dll...
WARN [xUnit.net 00:00:00.1314409] Skipping: TestAndPerf (Could not find any of the following assemblies: xunit.execution.desktop.dll)
INFO Discoverer finished.
INFO Test discovery finished.
Test discovery finished.
```

which is suggesting that i am missing ```xunit.execution.desktop.dll```, i am using ```xunit.runner.console``` and its working fine. Following are the pkgs which have been added

```
nuget xunit
nuget xunit.abstractions
nuget xunit.assert
nuget xunit.core
nuget xunit.extensibility.core
nuget xunit.extensibility.execution
nuget xunit.runner.console
```

**Comment by wahmedswl** [comment]
Hi, @axodox i was targetting .Net 4.5, now targetting .Net 4.6, and its working fine now. I found that from https://github.com/xunit/xunit/issues/1138.

I am closing this ticket now as xUnit is fully supported and working awesome :-)
